analyzers/src/SonarAnalyzer.Common/Rules/Hotspots/DeliveringDebugFeaturesInProductionBase.cs

```diff
@@ -40,17 +41,19 @@ public abstract class DeliveringDebugFeaturesInProductionBase<TSyntaxKind> : Tra
 
         protected abstract TrackerBase<TSyntaxKind, InvocationContext>.Condition IsInvokedConditionally();
 
+        protected abstract TrackerBase<TSyntaxKind, InvocationContext>.Condition IsInDevelopmentContext();
+
         protected DeliveringDebugFeaturesInProductionBase(IAnalyzerConfiguration configuration)
             : base(configuration, DiagnosticId, MessageFormat) { }
 
         protected override void Initialize(TrackerInput input)
         {
             var t = Language.Tracker.Invocation;
             t.Track(input,
-                t.MatchMethod(
-                    new MemberDescriptor(KnownType.Microsoft_AspNetCore_Builder_DeveloperExceptionPageExtensions, "UseDeveloperExceptionPage"),
-                    new MemberDescriptor(KnownType.Microsoft_AspNetCore_Builder_DatabaseErrorPageExtensions, "UseDatabaseErrorPage")),
-                t.ExceptWhen(IsInvokedConditionally()));
+                    t.MatchMethod(new MemberDescriptor(KnownType.Microsoft_AspNetCore_Builder_DeveloperExceptionPageExtensions, "UseDeveloperExceptionPage"),
+                                  new MemberDescriptor(KnownType.Microsoft_AspNetCore_Builder_DatabaseErrorPageExtensions, "UseDatabaseErrorPage")),
+                    t.And(t.ExceptWhen(IsInvokedConditionally()),
```

Comment:
You can remove the `t.And` to achieve the same. The `Track` method has `params` conditions that behaves as `and` mode.